File: C++_20Q/Q11.cpp
#include <iostream>

int calls = 0;

int fib(int n) {

  if (n <= 1) {
    calls += 1;
    return n;
  }

  else {
    calls += 1;
    std::cout << calls << std::endl;
    return (fib(n - 1) + fib(n - 2));
  }
}

int main() { std::cout << fib(10); }

File: Java_20Q/Q11.java
public class Q11 {

  public

  static int calls = 0;

  static int fib(int n) {
    if (n <= 1) {
      calls += 1;
      System.out.println(calls);
      return n;
    }

    else {
      calls += 1;
      System.out.println(calls);
      return fib(n - 1) + fib(n - 2);
    }
  }

  public static void main(String[] args) {
    System.out.println(fib(4));

  }
}
